client/lib/parsing.rs

```diff
@@ -244,6 +244,44 @@ fn args_from_simple_or_complex(
     }
 }
 
+macro_rules! stringify_if_not_empty {
+    ($e: expr) => {
+        if !$e.is_empty() {
+            format!("{}={}", stringify!($e), $e)
+        } else {
+            "".to_string()
+        }
+    };
+}
+
+macro_rules! params_with_values {
+    ($($x:expr),+ $(,)?) => (vec![$((stringify_if_not_empty!($x))),+])
+}
+
+/// Private macro for enforcing that there are no conflicting arguments, including missing a
+/// required one. This is intended to make debugging the parse_*_info methods easier for callers.
```

Comment:
I think this could use a little more explanation: something like saying that it returns an error if there isn't exactly one of the provided args which is non-empty?